src/pip/_internal/resolution/resolvelib/requirements.py

```diff
@@ -104,8 +118,8 @@ def name(self):
         # type: () -> str
         return self._candidate.name
 
-    def find_matches(self):
-        # type: () -> Sequence[Candidate]
+    def find_matches(self, is_root, constraints):
+        # type: (bool, Sequence[SpecifierSet]) -> Sequence[Candidate]
```

Comment:
Maybe add an assert here to make sure the constraints are empty. (I think they should?)

Reply:
I'll check that and add an assert.